application/backend/src/services/job_service.py

```diff
@@ -59,4 +64,33 @@ async def update_job_status(job_id: UUID, status: JobStatus, message: str | None
             updates: dict = {"status": status}
             if message is not None:
                 updates["message"] = message
+            progress_ = 100 if status is JobStatus.COMPLETED else progress
+            if progress_ is not None:
+                updates["progress"] = progress_
             await repo.update(job, updates)
+
+    @classmethod
+    async def stream_logs(cls, job_id: UUID | str):
+        from core.logging import get_job_logs_path
+
+        log_file = get_job_logs_path(job_id=job_id)
+        if not os.path.exists(log_file):
+            raise ResourceNotFoundException(resource_id=job_id, resource_name="job_logs")
+
+        async def is_job_still_running():
+            job = await cls.get_job_by_id(job_id=job_id)
+            return job.status == JobStatus.RUNNING
+
+        async with await anyio.open_file(log_file) as f:
+            while True:
+                line = f.readline()
+                still_running = await is_job_still_running()
+                if not line:
+                    # wait for more lines if job is still running
+                    if still_running:
+                        await asyncio.sleep(0.5)
+                        continue
+                    # No more lines are expected
+                    else:
+                        break
+                yield line
```

Comment:
I'm wondering if maybe we could yield multiple lines at a time. This might make the endpoint faster when the user opens an old log or a very long running log that has thousands of logs.

I don't mind postponing that though until we have a UI integration, to avoid any premature optimization that might not be necessary.